### ISSUE TITLE
only handsontable/vue performance issue

### ISSUE THREAD

**Issue by k3341095** [handsontable/vue-handsontable-official#76]
I used to be handsontable and I was well.
I am using handsontable / vue in nuxtjs after the project has been changed to vue.

Existing handsontable used jquery and was very fast. 3000 row over
But the handsontable / vue is very slow.

For example, when you click the checkbox, you can click on it quickly.
Chrome shows a warning.
[Violation] Forced reflow while executing JavaScript took 89ms
[Violation] JavaScript took 361ms
[Violation] Forced reflow while executing JavaScript took 1131ms
[Violation] JavaScript took 253ms
[Violation] Forced reflow while executing JavaScript took 113ms
[Violation] JavaScript took 54ms
[Violation] Forced reflow while executing JavaScript took 123ms
[Violation] Forced reflow while executing JavaScript took 378ms
[Violation] Forced reflow while executing JavaScript took 447ms

Existing jquery is no longer ..
It works very well and fast ..
From more data

Can I find out why?

Putting it in data binding data () or putting it in computed, or the result is equally slow.

Is there a solution?

        height: "700",
                colHeaders: true,
                rowHeaders: true,
                manualColumnMove: true,
                manualColumnResize: true,
                autoWrapCol: false,
                autoWrapRow: false,
                currentRowClassName: 'htCurrentRow',
                viewportRowRenderingOffset: 70,

It is the only default option. You have not added any renderers.

upload movie

https://youtu.be/sLwEluueORI

**Comment by k3341095** [comment]
I tested it in the vue.js environment, not nuxt.
Slightly faster than nuxt.

[Violation] Forced reflow while executing JavaScript took 31ms
handsontable.js?4d38:2210 [Violation] 'keydown' handler took 201ms
[Violation] Forced reflow while executing JavaScript took 32ms
handsontable.js?4d38:2210 [Violation] 'keydown' handler took 162ms
[Violation] Forced reflow while executing JavaScript took 31ms

**Comment by jansiegel** [comment]
@k3341095, this might be connected to #50. Currently, if you're using an array of objects as your dataset, the wrapper calls the Handsontable's `updateSettings` method every time a cell value is changed, which can be really heavy on performance.
After #50 is fixed (it should be done in the upcoming release), the situation might get better.

**Comment by k3341095** [comment]
I hope it was the same problem.
I am pleased to be included in this release.

I hope to see the next release soon.
Thank you.

**Comment by k3341095** [comment]
> @k3341095, this might be connected to #50. Currently, if you're using an array of objects as your dataset, the wrapper calls the Handsontable's `updateSettings` method every time a cell value is changed, which can be really heavy on performance.
> After #50 is fixed (it should be done in the upcoming release), the situation might get better.

Something strange happened.
The slow phenomenon of nuxt.js has been resolved.

nuxt.js asyncData has been changed.

Mounted.

When you import data into asyncData, it is included in html innerhtml.

However, if you change the data property using the mounted or method, it will not be included in .. html.

Could it be related to this ... It is very fast now. same jquery

**Comment by jansiegel** [comment]
@k3341095, We just released version `3.0.0` of the wrapper. Could you check if it helps with your problem?

**Comment by k3341095** [comment]
Has the usage changed in nuxt.js?
I get an error.
[Vue warn]: Failed to mount component: template or render function not defined.

https://github.com/handsontable/vue-handsontable-official#installation 
The examples on this site also get errors.

It will not work on existing projects.
and nuxt.js v2.0 out!!

nuxt.js + handsontable/vue 3.0  + example
render function or template not defined in component: anonymous 
error!!

nuxt.js bug??? or handsontable bug?

code :
```
<template>
  <div id="hot-preview">
<no-ssr>
    <HotTable :settings="settings"></HotTable>
</no-ssr>
  </div>
</template>

<script>
if (process.browser) {
    var HotTable = require('@handsontable/vue')
}

  export default {
    data: function() {
      return {
        settings: {
          data: [
            ["", "Ford", "Volvo", "Toyota", "Honda"],
            ["2016", 10, 11, 12, 13],
            ["2017", 20, 11, 14, 13],
            ["2018", 30, 15, 12, 13]
          ],
          colHeaders: true,
          rowHeaders: true,
        }
      };
    },
    components: {
      HotTable
    }
  }
</script>

<style src="../node_modules/handsontable/dist/handsontable.full.css"></style>
<style>
  #test-hot {
    width: 600px;
    height: 400px;
    overflow: hidden;
  }
</style>
```

**Comment by k3341095** [comment]
nuxt.js 1.4.2 and 2.0 both 

working well handsontable/vue 2.0
but 3.0 not working
[Vue warn]: Failed to mount component: template or render function not defined.

render function or template not defined in component: anonymous
solved no-ssr tag input~

**Comment by wojciechczerniak** [comment]
Hi @k3341095,

I've replicated the error but turned out this Nuxt.js error is very generic. It's thrown if any JS error is encountered. There is an issue on Nuxt.js Github to cache previous error and display the original one. Sometimes I got the right one displayed. In my case it was missing `navigator` property, so I figured out that my SSR setup is wrong.

I've managed to run Handsontable in Nuxt.js 2.0.0 by following this article: https://medium.com/@codebeast_/why-your-third-party-plugin-dont-work-in-nuxt-and-how-to-fix-it-d1a8caadf422
